src/AbstractSQLRules2SQL.ts

```diff
@@ -889,6 +891,14 @@ const typeRules: Dictionary<MatchFn> = {
 		const replacement = TextValue(getAbstractSqlQuery(args, 2), indent);
 		return `REPLACE(${str}, ${find}, ${replacement})`;
 	},
+	ParseJSON: (args, indent) => {
+		checkArgs('ParseJSON', args, 1);
+		if (engine !== Engines.postgres) {
+			throw new SyntaxError('ParseJSON not supported on: ' + engine);
+		}
+		const column = TextValue(getAbstractSqlQuery(args, 0), indent);
+		return `${column}::jsonb ->> 0`;
+	},
```

Comment:
Same thing here, this isn't strictly parsing JSON but rather extracting an elememt as text, so eg `->` is `ExtractJSON` and `->>` is `ExtractJSONText` (the extra `>` in all these functions means to return text instead of JSON). But I found an even better way which is to use `#>>'{}'`. Overall I think it'd be best to do an
```
['ExtractJSONPath', JSONNode, path: TextNode]
```
although I'm not sure on
```
['ExtractJSONPath', JSONNode, path: TextNode, returnText?: boolean]
```
or
```
['ExtractJSONPathAsText', JSONNode, path: TextNode]
```
I'm leaning towards the latter with `ExtractJSONPathAsText` because it makes things like typechecking easier because
```
`ExtractJSONPathAsTextNode` would be a `TextNode` and `ExtractJSONPath` would be a `JSONNode` but the alternative is that `ExtractJSONPath extends [any, any, any, true]` is a `TextNode` but `ExtractJSONPath extends [any, any, any, false]` is a `JSONNode` and I expect it'll make things a pain

Reply:
I agree with being specific with the types, `ExtractJSONPathAsText` in this case. There will probably be more cases in the future where we might want `ExtractJSONPathAsJSON`, `ExtractJSONPathAsNumber`, etc. I'm thinking it would be pretty difficult to try and cover all cases with a single `ExtractJSONPath` and makes sense to manage return types at the function layer and not a number of extra parameters.

Reply:
@Page- Taking another look at the [Postgres docs](https://www.postgresql.org/docs/14/functions-json.html), I think we should just go ahead and call `json_extract_path_text()`. According to the docs, its equivalent to the `#>>` operator. Also the name makes it more obvious/explicit as to what we're doing, almost exactly the same name as our wrapper here.

Reply:
Sounds like we might should just call `json_extract_path_text()` if we're allowing callers to specify a path.

Reply:
Although `json_extract_pah_text()` seems to require a path passed in while that won't always be the case for us. In your example above, I'm think it should be `['ExtractJSONPathAsText', JSONNode, path?: TextNode]`, with the `path` being optional instead of required. When not provided, the entire JSON is returned as text.